src/main/java/com/google/sps/data/User.java

```diff
@@ -1,57 +1,198 @@
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Type;
 import java.util.ArrayList;
 import java.util.List;
 
-// Class representing a User
 public class User {
-  private ArrayList<Long> cookbook;
-  private ArrayList<Long> planner;
-  private ArrayList<Long> userRecipes;
-  private String email;
-  private String id;
-  private String username;
-  // shopping list will not be implented until after MVP
-  private ArrayList<String> shoppingList;
+  // API variables:
+  // client ID is generated from Ali's google APIs credentials page.
+  private static String CLIENT_ID = "1034390229233-u07o0iaas2oql8l4jhe7fevpfsbrtv7n.apps.googleusercontent.com";
+  // Payload class contains decrypted user information.
+  private Payload payload;
+  private Entity entity;
 
-  public User(String id, String email, String username) {
-    this.id = id;
-    this.email = email;
-    this.username = username;
-    planner = new ArrayList<Long>();
-    cookbook = new ArrayList<Long>();
-    userRecipes = new ArrayList<Long>();
+  /**
+    // User Entity Properties:
+
+    "displayName" : "Example Name"
+    "cookbook" : [ "longId1", "longId2", "longId3" ]
+    "userRecipes" : [ "longId1", "longId2", "longId3" ]
+    "planner" : [ "longId1", "longId2", "longId3" ]
+    "shoppingList" : [ "longId1", "longId2", "longId3" ]
+
+    // Properties that can be accessed from Payload object:
+
+    // These seven fields are only included when the user has granted the "profile" and
+    // "email" OAuth scopes to the application.
+    // "sub" or Subject refers to the user-unique ID number.
+    "sub" : "110169484474386276334"
+    "email" : "testuser@gmail.com",
+    "email_verified" : "true",
+    "name" : "Test User",
+    "picture" : "https://lh4.googleusercontent.com/-kYgzyAWpZzJ/ABCDEFGHI/AAAJKLMNOP/tIXL9Ir44LE/s99-c/photo.jpg",
+    "given_name" : "Test",
+    "family_name" : "User",
+  */
+
+  /**
+   * Used for initialization with static methods.
+   */
+  private User() { }
+
+  /**
+   * Constructs a User instance based on a unique id_token.
+   * Verify user using google's ID token verifyer API.
+   * This method is commented out because it does not work yet.
+   */
+  public User(String idTokenString) {
+    // GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier
+    // .Builder(UrlFetchTransport.getDefaultInstance(), new JsonFactory())
+    // // Specify the CLIENT_ID of the app that accesses the backend:
+    // .setAudience(Collections.singletonList(CLIENT_ID)).build();
+
+    // GoogleIdToken idToken = verifier.verify(idTokenString);
+    // if (idToken == null) {
+    //   throw new Exception("ID token verification failed");
+    // }
+
+    // payload = idToken.getPayload();
+  }
+
+  /**
+   * This creates a standard universal user instance regardless of login status.
+   * This test method will not be removed from the webapp prior to deployment.
+   */
+  public static User getTestUser() {
```

Comment:
It seems a little weird to return an instance of the object by a static method..can you make this a private helper method wherever you need to use it? Alternatively, you could move this logic to the constructor for now? Either use the constructor with no parameters, or have a boolean parameter indicating whether it's a test user or not.